projects/packages/autoloader/src/class-container.php

```diff
@@ -60,8 +60,8 @@ private function register_shared_dependencies() {
 		}
 
 		$key = self::SHARED_DEPENDENCY_KEYS[ Hook_Manager::class ];
+		require_once __DIR__ . '/class-hook-manager.php';
```

Comment:
Hi @reatang! Thanks for the PR.

The `Container` class only needs to load the `class-hook-manager.php` file when the condition in the if statement is true. So, I'm not sure that it makes sense to always load the `class-hook-manager.php` file here.

Also, I don't think you're encountering a bug in the `jetpack-autoloader` package.

It looks like you would like to use an object of the `Hook_Manager` class in your plugin. If your plugin needs to use a file that isn't already loaded, it's probably best to have the plugin load the file it needs. Is there a reason that your plugin needs to use a `Hook_Manager` object from a specific namespace?

Reply:
Thank you for reviewing this pull request.
if each plugin requires a unique Hook_Manager class, then that class should be loaded multiple times, not just one. I think that's what Jeckack-autoload `specific namespace` was designed for.

Reply:
Hi @reatang! I don't see any reason for the `Hook_Manager` class to be loaded multiple times. The single `Hook_Manager` object is created in the `Container::register_shared_dependencies` method. In that method, the object [is stored](https://github.com/Automattic/jetpack/blob/master/projects/packages/autoloader/src/class-container.php#L63-L66) in the `$jetpack_autoloader_container_shared` global variable, and it's used by all instances of the `jetpack-autoloader`.

The code below is from the `Container::register_shared_dependencies` method. It shows that the `$jetpack_autoloader_container_shared` global is checked first, before a `Hook_Manager` object is created. So, the `class-hook-manager.php` file needs to be loaded only when the `Hook_Manager` object hasn't been created yet:

```
if ( ! isset( $jetpack_autoloader_container_shared[ $key ] ) ) {
	require_once __DIR__ . '/class-hook-manager.php';
	$jetpack_autoloader_container_shared[ $key ] = new Hook_Manager();
}
```